d8error.py

```diff
@@ -23,24 +30,33 @@ def __init__(self, etype, value):
                 if (key in str(value)):
                     prewrittenMessge = True
             self.print = prewrittenMessge
-            
+
+        # save errors to errorLog.csv
+        def writeRow(file):
+            fieldnames = ['index', 'errorType', 'errorMSG', 'feedbackRating', 'feedbackMSG']
+            writer = csv.DictWriter(file, fieldnames=fieldnames)
+            writer.writerow({"index": self.eindex,
+                            "errorType": self.errorname,
+                            "errorMSG": str(self.value),
+                            "feedbackRating": self.feedbackRating,
+                            "feedbackMSG": self.feedbackMSG})
+
         if not os.path.isfile("errorLog.csv"):
             with open('errorLog.csv', 'w', newline='') as f:
-                
-                fieldnames = ['errorType', 'errorMSG']
-                writer = csv.DictWriter(f, fieldnames=fieldnames)
-                writer.writerow({"errorType": self.errorname,"errorMSG": str(value)})
+                writeRow(f)
         else:
+            if Announce.eindex == 1:
+                with open("errorLog.csv", 'r') as f:
+                    for row in reversed(list(csv.reader(f))):
```

Comment:
Isn't calling `reversed` equally expensive as iterating through the whole reader?